Tools/android/GenerateAndroidMk.py

```diff
@@ -0,0 +1,224 @@
+
+    def getDefines(self, name):
+        flags = self.getFlagsfile()
+        with open(flags) as f:
+            text = f.read()
+        line = re.findall(name + " =.*\n", text)[0]
+        line = line.replace(name + " = ", "")
+        line = INDENT + line.replace(" ", " \\\n" + INDENT)
+        return line
+
+    def getSources(self):
+        makefile = self.getMakefile()
+        with open(makefile) as f:
+            text = f.read()
+        lines = re.findall(".*?\\.o:\\n", text)
+        lines = [l.replace(".o:\n", " \\") for l in lines]
+        self.adjustSources(lines)
+        #make source pretty
+        return INDENT + ("\n" + INDENT).join(lines)
+
+
+class GmmGeneator(Generator):
+    def __init__(self, root):
+        src = path.join(root, "gmmlib")
+        templ = path.join(src,  "Tools/android")
+        super(GmmGeneator, self).__init__(src, templ)
+
+    def getTemplatePath(self):
+        return path.join(self.templ_, "gmm.Android.mk.template")
+
+    def getMakefile(self):
+        return path.join(self.build_, "Source/GmmLib/Makefile")
+
+    def getFlagsfile(self):
+        return path.join(self.build_, "Source/GmmLib/CMakeFiles/gmm_umd.dir/flags.make")
+
+    def adjustSources(self, lines):
+        for i, l in enumerate(lines):
+            j = l.find("__/")
+            if j == -1:
+                lines[i] = path.join("Source/GmmLib", l)
+            else:
+                lines[i] = path.join("Source", l[j+3:])
+
+
+class CmrtGeneator(Generator):
+    def __init__(self, root):
+        src = path.join(root, "media-driver/cmrtlib")
+        templ = path.join(src,  "android")
+        super(CmrtGeneator, self).__init__(src, templ)
+
+    def getTemplatePath(self):
+        return path.join(self.templ_, "cmrt.Android.mk.template")
+
+    def getMakefile(self):
+        return path.join(self.build_, "linux/Makefile")
+
+    def getFlagsfile(self):
+        return path.join(self.build_, "linux/CMakeFiles/igfxcmrt.dir/flags.make")
+
+    def adjustSources(self, lines):
+        for i, l in enumerate(lines):
+            j = l.find("__/")
+            if j == -1:
+                lines[i] = path.join("linux", l)
+            else:
+                lines[i] = l[j+3:]
+
+
+class DriverGeneator(Generator):
+    def __init__(self, root):
+        src = path.join(root, "media-driver/media_driver")
+        templ = path.join(src,  "android")
+        super(DriverGeneator, self).__init__(src, templ)
+
+    def getCmakeCmd(self):
+        """
+        cmake ../../../../media-driver -DCMAKE_INSTALL_PREFIX=/usr -DMEDIA_VERSION="2.0.0" \
+        -DBUILD_ALONG_WITH_CMRTLIB=1 -DBS_DIR_GMMLIB=`pwd`/../../../../gmmlib/Source/GmmLib/ \
+        -DBS_DIR_COMMON=`pwd`/../../../../gmmlib/Source/Common/ -DBS_DIR_INC=`pwd`/../../../../gmmlib/Source/inc/ \
+        -DBS_DIR_MEDIA=`pwd`/../../../../media-driver
+        """
+        cmd = 'cmake ../../../../media-driver -DCMAKE_INSTALL_PREFIX=/usr -DMEDIA_VERSION="2.0.0" '
```

Comment:
I tried to generate cmake for media_driver only. but it failed and complain about missing "media_defs.cmake"
could we fix it ?

step to reproduce:
1. cd media-driver/media_dirver/build
2. cmake ..

```
CMake Error at CMakeLists.txt:107 (include):
  include could not find load file:

    /sdb/android/1a0830/vendor/intel/video/media-driver/Source/media/media_driver/cmake/media_defs.cmake


CMake Error at CMakeLists.txt:109 (include):
  include could not find load file:

    /media_utils.cmake


CMake Error at media_top_cmake.cmake:26 (include):
  include could not find load file:

    /media_gen_flags.cmake
Call Stack (most recent call first):
  CMakeLists.txt:111 (include)


CMake Error at media_top_cmake.cmake:27 (include):
  include could not find load file:

    /media_feature_flags.cmake
Call Stack (most recent call first):
  CMakeLists.txt:111 (include)

```

Reply:
will look at this

Reply:
@oliver-sang , any update on this
thanks